### ISSUE TITLE
Deletion commands do not work via fetch

### ISSUE THREAD

**Issue by CraigglesO** [stripe/stripe-node#1360]
NodeJS version: v16.13.2
Stripe version: ^8.197.0

I am using a cloudflare worker, so if you check the webpack config [here](https://github.com/stripe-samples/stripe-node-cloudflare-worker-template) you can see it is limiting stripe to fetch and building like so:

```js
const STRIPE = Stripe(STRIPE_API_CODE, { httpClient: Stripe.createFetchHttpClient() })
```

commands like `await STRIPE.customers.deleteTaxId` and `await STRIPE.customers.del` do not work. An example error is as follows:

```
Zo [Error]: An error occurred with our connection to Stripe.
    at /Users/ct/Documents/Projects/node_modules/.pnpm/stripe@8.202.0/node_modules/stripe/lib/StripeResource.js:519:15
    at processTicksAndRejections (node:internal/process/task_queues:96:5) {
  type: 'Zo',
  raw: {
    message: 'An error occurred with our connection to Stripe.',
    detail: TypeError: fetch failed
        at Object.processResponse (/Users/ct/.npm/_npx/dce38835dbc726fc/node_modules/undici/lib/fetch/index.js:183:23)
        at Fetch.fetchFinale (/Users/ct/.npm/_npx/dce38835dbc726fc/node_modules/undici/lib/fetch/index.js:937:17)
        at Fetch.mainFetch (/Users/ct/.npm/_npx/dce38835dbc726fc/node_modules/undici/lib/fetch/index.js:744:17)
        at processTicksAndRejections (node:internal/process/task_queues:96:5) {
      cause: [RequestContentLengthMismatchError]
    }
  },
  rawType: undefined,
  code: undefined,
  doc_url: undefined,
  param: undefined,
  detail: TypeError: fetch failed
      at Object.processResponse (/Users/ct/.npm/_npx/dce38835dbc726fc/node_modules/undici/lib/fetch/index.js:183:23)
      at Fetch.fetchFinale (/Users/ct/.npm/_npx/dce38835dbc726fc/node_modules/undici/lib/fetch/index.js:937:17)
      at Fetch.mainFetch (/Users/ct/.npm/_npx/dce38835dbc726fc/node_modules/undici/lib/fetch/index.js:744:17)
      at processTicksAndRejections (node:internal/process/task_queues:96:5) {
    cause: RequestContentLengthMismatchError: Request body length does not match content-length header
        at write (/Users/ct/.npm/_npx/dce38835dbc726fc/node_modules/undici/lib/client.js:1432:37)
        at _resume (/Users/ct/.npm/_npx/dce38835dbc726fc/node_modules/undici/lib/client.js:1384:29)
        at resume (/Users/ct/.npm/_npx/dce38835dbc726fc/node_modules/undici/lib/client.js:1243:3)
        at processTicksAndRejections (node:internal/process/task_queues:82:21) {
      code: 'UND_ERR_REQ_CONTENT_LENGTH_MISMATCH'
    }
  },
  headers: undefined,
  requestId: undefined,
  statusCode: undefined,
  charge: undefined,
  decline_code: undefined,
  payment_intent: undefined,
  payment_method: undefined,
  payment_method_type: undefined,
  setup_intent: undefined,
  source: undefined
}
```

I believe `RequestContentLengthMismatchError` is the key to this error.

**Comment by jnordberg** [comment]
I'm also running into something similar to this, not for delete requests though. I think the issue is that the stripe lib sets content-length to 0 for empty requests and the fetch implementation by Cloudflare uses [undici](https://undici.nodejs.org) which [doesn't allow that](https://github.com/nodejs/undici/blob/cedc7d26f64aaa0571d1af6eaf82c519a1bcc6da/lib/client.js#L1420-L1436). Not sure if the Cloudflare fetch wrapper is doing something wrong or most other http client implementations choose to fix that for you instead of bombing like undici does, it's a bit silly IMO.

I think this has to be addressed in wrangler or wherever that fetch > undici bridge lives

**Comment by jnordberg** [comment]
Seems this has been fixed in miniflare 2.4.0 (not published to npm yet, probably will be soon): https://github.com/cloudflare/miniflare/issues/193

**Comment by CraigglesO** [comment]
Just incase stripe is ok with this fix:
https://github.com/stripe/stripe-node/pull/1370
